eng/pipelines/runtime-sanitized.yml

```diff
@@ -1,14 +1,106 @@
 # This pipeline provides an easy mechanism for us to run runtime and libaries tests with native sanitizers enabled
-# without having to compilcate the runtime.yml pipeline. This pipeline is intended to be run on a rolling basis
-# but a schedule is yet to be determined.
+# without having to compilcate the runtime.yml pipeline.
 trigger: none
 
+schedules:
+- cron: "0 11 * * 2"
+  displayName: Tuesday at 3:00 AM (UTC-8:00)
+  branches:
+    include:
+    - main
+
 variables:
   - template: /eng/pipelines/common/variables.yml
 
 extends:
   template:  /eng/pipelines/common/templates/pipeline-with-resources.yml
   parameters:
     stages:
-    - stage: Build
+    - stage: AddressSanitizer
       jobs:
+      #
+      # Build the whole product with CoreCLR and run runtime tests with AddressSanitizer
+      #
+      - template: /eng/pipelines/common/platform-matrix.yml
+        parameters:
+          jobTemplate: /eng/pipelines/common/global-build-job.yml
+          helixQueuesTemplate: /eng/pipelines/coreclr/templates/helix-queues-setup.yml
+          buildConfig: Checked
+          runtimeFlavor: coreclr
+          platforms:
+            - linux_x64
+            - osx_x64
+          variables:
+            - name: _nativeSanitizersArg
+              value: -fsanitize address
+          jobParameters:
+            nameSuffix: CoreCLR_RuntimeTests
+            buildArgs: -s clr+libs -c $(_BuildConfig) $(_nativeSanitizersArg)
+            timeoutInMinutes: 300
+            # extra steps, run tests
+            extraStepsTemplate: /eng/pipelines/common/templates/runtimes/build-runtime-tests-and-send-to-helix.yml
+            extraStepsParameters:
+              creator: dotnet-bot
+              testRunNamePrefixSuffix: CoreCLR_$(_BuildConfig)
+              scenarios:
+                - normal
+                - no_tiered_compilation
+            extraVariablesTemplates:
+              - template: /eng/pipelines/common/templates/runtimes/test-variables.yml
+                parameters:
+                  testGroup: outerloop
+                  sanitizers: 'address'
+
+      #
+      # Build the whole product with CoreCLR and run libraries tests with AddressSanitizer
+      #
+      - template: /eng/pipelines/common/platform-matrix.yml
+        parameters:
+          jobTemplate: /eng/pipelines/common/global-build-job.yml
+          helixQueuesTemplate: /eng/pipelines/libraries/helix-queues-setup.yml
+          buildConfig: Debug
```

Comment:
Is there a reason to run the libraries tests as `Debug` as opposed to `Release`? It is highly unlikely any failure will result in something actionable from within managed code. The real interesting part here is forcing the Checked runtime to fully JIT optimize the libraries themselves.

Reply:
We get better ASAN coverage of the native libraries components (like System.Security.Cryptography.Native) if we build the libraries components in Debug. I agree that the managed code is uninteresting, but the native code definitely can be.